lambdas/build-cri-oauth-request/src/main/java/uk/gov/di/ipv/core/buildcrioauthrequest/BuildCriOauthRequestHandler.java

```diff
@@ -333,8 +351,45 @@ private JWEObject signEncryptJar(
                         evidenceRequest,
                         context);
 
-        RSAEncrypter rsaEncrypter = new RSAEncrypter(oauthCriConfig.getParsedEncryptionKey());
-        return AuthorizationRequestHelper.createJweObject(rsaEncrypter, signedJWT);
+        var encKey = getCriEncryptionKey(oauthCriConfig);
+        RSAEncrypter rsaEncrypter = new RSAEncrypter(encKey);
+        return AuthorizationRequestHelper.createJweObject(
+                rsaEncrypter, signedJWT, encKey.getKeyID());
+    }
+
+    private RSAKey getCriEncryptionKey(OauthCriConfig criConfig) throws ParseException {
```

Comment:
I think it's worth pulling this out into a separate `OauthKeyService` or similar - this class is already quite big (and does several things!)